x/nexus/keeper/transfer.go

```diff
@@ -84,14 +143,29 @@ func (k Keeper) EnqueueForTransfer(ctx sdk.Context, sender exported.CrossChainAd
 		return 0, fmt.Errorf("recipient's chain %s does not support foreign assets", recipient.Chain.Name)
 	}
 
+	// merging incomplete transfers for the specified recipient
+	incompleteTransfer, found := k.getTransferForRecipientAndAsset(ctx, recipient, asset.Denom, exported.Incomplete)
+	if found {
+		asset = asset.Add(incompleteTransfer.Asset)
+		k.deleteTransfer(ctx, incompleteTransfer)
+	}
```

Comment:
why do we need to differenciate between incomplete and pending transfers? Can't we just keep merging transfers until there is enough to cover the fee, like before?

Reply:
So, there were two reasons for this:
1. With a new min fee, now we need to distinguish between pending transfers that were below the minimum vs a transfer were the fee was already paid and the remaining amount is now below the minimum. For e.g. suppose base fee is `6`, if a transfer of `10` is sent, we will take `6` as fee and set a pending transfer of `4`. Whereas if a transfer of `4` is sent, being below the minimum, we will set it as a pending transfer of `4`. Now we can't distinguish between these two cases.
2. It's not very nice that the pending transfers below the minimum were being reprocessed everytime. So we wanted to assign them a separate status. But due to the way we store them in the keeper, checking if there's an incomplete transfer to the same recipient is still not as efficient. But this were point 3 might be a useful future improvement.
3. I don't like to keep around these small transfers in the keeper long term. I think it might be nice to have a message that just claims all these small transfers to the fee collector account if the user hasn't bothered to send more to the recipient. Alternatively, we always collect any transfer below the minimum as fees and make it user's responsibility to send a sufficient amount. We do end up paying when users try to do a burn with even a small amount.

Reply:
To point 3, maybe we could perform that periodically via the end blocker, instead of using a specific tx to explicitly collect the transfers as fees.

Reply:
there is however one thing that confuses me. Isn't this PR actually replacing the minimum amount for a specific fee per asset? Some of the commands have been changed to not accept a min amount anymore.

Reply:
I like the end blocker idea.

The minimum transfer fee (sum of min fee for source/destination chain for that asset) now acts as the minimum amount. If the transfer amount is equal to or below this, then that transfer is set as incomplete instead. Hence, the min amount checks elsewhere aren't needed.